org.OpenT2T.Sample.SuperPopular.Thermostat/Wink Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+var https = require('https');
+var wh = require("opent2t-translator-helper-wink"); //link locally for now
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+var deviceId, accessToken;
+
+// module exports, implementing the schema
+module.exports = {
+
+    device: null,
+
+    initDevice: function(dev) {
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.access_token) !== 'undefined') {
+                    accessToken = props.access_token;
+                } else {
+                    console.log('props.access_token is undefined.');
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
```

Comment:
For all these error cases, you should `throw new Error("...")` instead of just logging.